File: LC 866.cpp
class Solution {
public:
    bool isPrime(int n) {
        if (n < 2) return false;
        for (int i = 2; i <= sqrt(n); i++) {
            if (n % i == 0) return false;
        }
        return true;
    }

    int reverse(int n) {
        int reversed = 0;
        while (n > 0) {
            reversed = 10 * reversed + (n % 10);
            n /= 10;
        }
        return reversed;
    }

    bool isPalindrome(int n) {
        return n == reverse(n);
    }



    int primePalindrome(int n) {
        while (true) {
            if (isPalindrome(n) && isPrime(n)) return n;
            n++;
            if (n > 11 && to_string(n).size() % 2 == 0) {
                n = pow(10, to_string(n).size());
            }
        }
    }

};

File: 1000/LC 866.go
func primePalindrome(n int) int {
    for {
        if isPalindrome(n) && isPrime(n) {
            return n
        }
        n++
        s := strconv.Itoa(n)
        t := len(s)
        if n > 11 && t & 1 == 0 {
            n = quickPow(10, t)
        }
    }

    return n
}

func quickPow(a, r int) (res int) {
    res = 1
    for ; r > 0; r >>= 1 {
        if r & 1 == 1 {
            res *= a
        }
        a *= a
    }
    return 
}

func isPrime(n int) bool {
    if n < 2 {
        return false
    }
    for i := 2; i * i <= n; i++ {
        if n % i == 0 {
            return false
        }
    }

    return true
}

func reverse(n int) (ans int) {
    for n > 0 {
        ans = ans * 10 + n % 10
        n /= 10
    }
    return
}

func isPalindrome(n int) bool {
    return n == reverse(n)
}